### PR TITLE
Analyzer: Pack better

### LINKED ISSUE #11633: 2.1 preview2 - why so many packages
When moving my 4.6.1 based provider from preview1 to preview2, the number of packages has exploded - is this expected?

https://github.com/ErikEJ/EntityFramework.SqlServerCompact/commit/15a43663d3712b388530e90c518ce21259b98716

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #11633

### CODE REVIEW

**Comment by smitpatel** [comment]
Do you want to verify https://github.com/aspnet/EntityFrameworkCore/pull/11508#discussion_r178408265 also?

**Comment by bricelam** [comment]
No. The way it is now is very, very close to what the VS item template for analyzers generates.

File: src/EFCore.Analyzers/EFCore.Analyzers.csproj [hunk 1]
```diff
@@ -18,11 +18,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisCSharpWorkspacesPackageVersion)" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisCSharpWorkspacesPackageVersion)" PrivateAssets="all" />
+    <PackageReference Update="NETStandard.Library" PrivateAssets="all" />
```
Comment (divega) [inline src/EFCore.Analyzers/EFCore.Analyzers.csproj]: Will this (plus the fact that this targets .NETStandard 1.3) have an impact on how many packages are restored on .NET Framework?

Comment (divega) [reply src/EFCore.Analyzers/EFCore.Analyzers.csproj]: Ah, I missed that it exactly what this change is supposed to fix!

Comment (bricelam) [reply src/EFCore.Analyzers/EFCore.Analyzers.csproj]: Correct--it was already automatically referenced. This just adds `PrivateAssets="all"` to keep it out of the `*.nuspec`.

**Pull request merged**